### PR TITLE
Fragility Algo. Added As Analyzer

### PR DESCRIPTION
**Pull request by adam2392**
Added the fragility analysis algo. as an analyzer. The demo notebook is in corresponding pull request in tvb-documentation within demos directory.

This allows one to analyze the corresponding output time series generated by the TVB either in a serial fashion, or parallel (with multiprocessing pool).

### CODE REVIEW

**Comment by adam2392** [comment]
The corresponding notebook demo will be added to this directory when the tvb-documentation is merged. 

https://github.com/the-virtual-brain/tvb-documentation/tree/master/demos

Let me know if there is anything else I need to add.

File: tvb/analyzers/fragility.py
Comment (maedoc) [inline tvb/analyzers/fragility.py:74]: `psutil` is a soft dependency, please use `multiprocessing.cpu_count()`, which ships with Python

File: tvb/analyzers/fragility.py
Comment (maedoc) [inline tvb/analyzers/fragility.py]: single line between methods please

File: tvb/analyzers/fragility.py
Comment (maedoc) [inline tvb/analyzers/fragility.py]: there's a problem here: `tempdir` will be `None` not `str`

File: tvb/analyzers/fragility.py
Comment (maedoc) [inline tvb/analyzers/fragility.py]: as above, please use builtin `multiprocessing.cpu_count()`

File: tvb/analyzers/fragility.py
Comment (maedoc) [inline tvb/analyzers/fragility.py]: this can return just a simple dictionary liek `{"radius": self.radius}`

File: tvb/analyzers/fragility_lib/execute/singlecore/singlemvar.py
Comment (maedoc) [inline tvb/analyzers/fragility_lib/execute/singlecore/singlemvar.py]: modifying `sys.path` isn't appropriate; you're probably just missing some `__init__.py` files in the relevent folders.  Let me know if I should take a look

File: tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py
Comment (maedoc) [inline tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py]: `'temp_{0}.npz'.format(win)`

File: tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py
Comment (maedoc) [inline tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py]: `super(ParallelMvar, self).__init__(...)` is idiomatic because it handles multiple inheritance correctly.

File: tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py
Comment (maedoc) [inline tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py]: should this be a logging statement instead, e.g. 

`logger.debug('Finsih comp for %d windows', self.numwins)`

File: tvb/simulator/monitors.py
Comment (maedoc) [inline tvb/simulator/monitors.py]: isn't this change done in the other PR? if so, handle my comments there on that branhc and then we can rebase this one on that, before merging this PR

**Review by maedoc — CHANGES_REQUESTED** [review]
looks good, but please try to clean up the style per my comments also consider running autopep8 on the files you've added, and then we'll merge.

File: tvb/analyzers/fragility.py
Comment (adam2392) [reply tvb/analyzers/fragility.py:74]: Fixed.

File: tvb/analyzers/fragility.py
Comment (adam2392) [reply tvb/analyzers/fragility.py]: Fixed.

File: tvb/analyzers/fragility.py
Comment (adam2392) [reply tvb/analyzers/fragility.py]: Fixed.

File: tvb/analyzers/fragility.py
Comment (adam2392) [reply tvb/analyzers/fragility.py]: Added return statements for both the linear model estimation and the perturbation model.

File: tvb/simulator/monitors.py
Comment (adam2392) [reply tvb/simulator/monitors.py]: Yes, I think I made a mistake when I was modifying files on this branch. I reverted the file back to what is on the trunk of the master branch. 

I updated the other branch (obs-noise) with updates to this.

File: tvb/analyzers/fragility.py
Comment (adam2392) [reply tvb/analyzers/fragility.py]: Fixed assertion statement.

File: tvb/analyzers/fragility_lib/execute/singlecore/singlemvar.py
Comment (adam2392) [reply tvb/analyzers/fragility_lib/execute/singlecore/singlemvar.py]: Yes, you are right. I removed this line. I think the 

"from ...linearmodels.basemodel ..." 

should handle it correctly.

File: tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py
Comment (adam2392) [reply tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py]: Fixed these string formatters.

File: tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py
Comment (adam2392) [reply tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py]: Updated to be super(<classname>, self) for each file in the execute dir.

**Pull request closed without merging**
